src/gatsby/plugins/gatsby-plugin-openapi/types.ts

```diff
@@ -0,0 +1,89 @@
+export type RequestBodySchema = {
+  required: string[];
+  type: string;
+  properties: {
+    [key: string]: { type: string; description: string; example: string };
+  };
+};
+
+
+export type Parameter = {
+  schema: {
+    type: string;
+    format: string;
+    enum: string[];
+  };
+  name: string;
+  in: string;
+  description: string;
+  required: boolean;
+};
+
+export type DeRefedOpenAPI = {
+  paths: {
+    [key: string]: {
+      [key: string]: {
+        operationId: string;
+        tags: string[];
+        parameters: Parameter[];
+        requestBody: {
+          content: {
+            "application/json": {
+              schema: RequestBodySchema;
+            };
+          };
+          required: boolean;
+        };
+        responses: {
+          [key: string]: {
+            description: string;
+            content: {
+              "application/json": {
+                schema: any;
+                example: any;
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+};
+
+export type ResponseContent = {
+  content_type: string;
```

Comment:
@NisanthanNanthakumar from the code, it looked like `content-type` should have a dash and not an underscore.

Reply:
@scefali 👍 apparently GraphQL fields cannot have hyphens, so I'll covert & keep it with the underscore.